api/src/middlewares/webhookMiddleware.ts

```diff
@@ -142,6 +175,40 @@ const webhookMiddleware = async (req, res, next) => {
       conversationMessageInserted: message
     });
 
+    // company
+    let company = await findCompany(params);
+
+    const hasCompanyFields = await checkCompanyFieldsExists(params);
+
+    if (hasCompanyFields) {
+      const companyDoc = {
+        primaryEmail: params.companyPrimaryEmail,
+        primaryPhone: params.companyPrimaryPhone,
+        primaryName: params.companyPrimaryName,
+        website: params.companyWebsite,
+        industry: params.companyIndustry,
+        businessType: params.companyBusinessType,
+        avatar: params.companyAvatar
+      };
+
+      if (!company) {
+        company = await Companies.createCompany(companyDoc);
+      }
+
+      company = await Companies.updateCompany(company._id, companyDoc);
```

Comment:
if the "if" condition above is valid, then the company would be updated with a same doc, put else